### PR TITLE
Add docs for mix help

### PR DESCRIPTION
**Pull request by pinfieldharm**
- Add shortdoc so that mix -h shows task
- Add moduledoc so that mix help test.watch has something

### CODE REVIEW

File: lib/mix_test_watch.ex [hunk 1]
```diff
@@ -2,6 +2,11 @@ defmodule Mix.Tasks.Test.Watch do
   use Mix.Task
   use GenServer
 
+  @shortdoc "Auto-run tests"
+
+  @moduledoc """
+    A task for running tests whenever source files change.
```
Comment (lpil) [inline lib/mix_test_watch.ex]: Is there supposed to be 2 spaces at the start of this line?

**Comment by lpil** [comment]
This is great, thank you! Please see the comment above.

**Comment by pinfieldharm** [comment]
Not sure how those extra spaces ended up in there, removed.

**Comment by lpil** [comment]
Thank you! :)

**Pull request merged**